libr/io/map.c

```diff
@@ -157,10 +157,11 @@ R_API bool r_io_map_remap (RIO *io, ut32 id, ut64 addr) {
 	if (!(map = r_io_map_resolve (io, id))) {
 		return false;
 	}
+	ut64 size = map->itv.size;
 	map->itv.addr = addr;
-	if (UT64_MAX - map->itv.size + 1 < addr) {
-		r_io_map_new (io, map->fd, map->flags, map->delta - addr, 0, map->itv.size + addr, true);
-		map->itv.size = -map->itv.addr;
+	if (UT64_MAX - size + 1 < addr) {
```

Comment:
why do you change the order here

Reply:
r_io_map_new(............, true) recalculates skyline. The change to `map->itv.size` should happen before recalculation.